core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -467,7 +474,7 @@ private[spark] class TaskSchedulerImpl private[scheduler](
       taskState: TaskState,
       reason: TaskFailedReason): Unit = synchronized {
     taskSetManager.handleFailedTask(tid, taskState, reason)
-    if (!taskSetManager.isZombie && taskState != TaskState.KILLED) {
+    if (!taskSetManager.isZombie) {
```

Comment:
@ericl Actually that is not correct.
Killed tasks were not candidates for resubmission on failure; and hence there is no need to revive offers when task kills are detected.

If they are to be made candidates, we need to introduce this expectation explicit elsewhere also to be consistent.